### PR TITLE
Cleanup I/O Handling

### PR DESCRIPTION
**Pull request by RobertBColton**
This was duplicated all over with the same logic.

### CODE REVIEW

File: ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp [hunk 1]
```diff
@@ -569,19 +569,6 @@ int window_get_cursor()
...
 
-void io_handle()
-{
-  MSG msg;
-  enigma::input_push();
-  while (PeekMessage (&msg, NULL, 0, 0, PM_REMOVE))
-  {
-    TranslateMessage (&msg);
-    DispatchMessage (&msg);
-  }
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp:580]: I don't think this ever worked because it's not actually running our window procedure.

**Comment by RobertBColton** [comment]
Yeah, I'm not, we can wait for @time-killer-games or somebody to test and see.

**Comment by fundies** [comment]
I think josh was trying to con you into writing a test

**Comment by RobertBColton** [comment]
Oh, if that's what he meant, I think I should resolve #2040 first. Our input handling seems to be a tad off with how GM handles ALT+TAB and we are actually handling mouse and keyboard input different. For the keyboard we push the current and last state on a window message, but for mouse we only push current state and then later update last state.

I'm fine to write a test but the other stuff maybe needs cleaned up first (there's 2041 too).

**Comment by RobertBColton** [comment]
Alright, so I gave this a little test in GM8.1 and it does not allow you to quit.
```cpp
while(true) {
    io_handle();
    draw_clear(c_white);
    draw_text(0,0,"wtf");
    screen_refresh();
}
```

File: ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp [hunk 2]
```diff
@@ -632,15 +632,6 @@ void initkeymap() {
 
 namespace enigma_user {
 
-void io_handle() {
-  enigma::input_push();
-  while (XQLength(disp)) {
-    DEBUG_MESSAGE("processing an event...", MESSAGE_TYPE::M_INFO);
-    if (enigma::handleEvents() > 0) exit(0);
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp:639]: X11 was already calling the full thing @JoshDreamland

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp:639]: Also there really is no such thing as full or partial event loops, TranslateMessage always calls the same window procedure. All platforms are the same in this regard.

**Comment by RobertBColton** [comment]
There 366f7f20b3371c442baa558d9931e8304aa754ff I disabled the quit like GM8.1 and Windows on master. Now the only difference here with master is that Xlib and SDL do not allow quit either. However, small thing, both the master version and this pull request freeze my Ubuntu virtual machine and my Windows desktop under SDL.

File: ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp (hunk 2 above)
Comment (JoshDreamland) [reply ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp:639]: Does handleEvents already loop until the queue is completely empty? If that's the case, this change should be fine.

**Comment by RobertBColton** [comment]
Yes @JoshDreamland handleEvents draining the queue that be the case on all platforms and xlib.

**Comment by RobertBColton** [comment]
Ha, GM8.1 allows the close button to interrupt `keyboard_wait();` calls. So that's perfect, ENIGMA behaves the same now, it looks like everything is good to go when this passes.
```cpp
keyboard_wait();
show_message("hello");
```

File: ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp (hunk 2 above)
Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp:639]: Yes @JoshDreamland handleEvents draining the queue that be the case on all platforms and xlib.

**Pull request merged**